src/tr2/game/objects/creatures/monk.c

```diff
@@ -259,5 +249,24 @@ static void M_Control(const int16_t item_num)
     Creature_Animate(item_num, angle, 0);
 }
 
-REGISTER_OBJECT(O_MONK_1, M_Setup1)
+void Monk1_Setup(OBJECT *const obj)
+{
+    if (!obj->loaded) {
+        return;
+    }
+    M_SetupBase(obj);
+    obj->pivot_length = 0;
+}
+
+void Monk3_Setup(OBJECT *const obj)
+{
+    if (!obj->loaded) {
+        return;
+    }
+    Monk1_Setup(obj);
```

Comment:
Let's call here the base function. I really want us to avoid inter-object setup calls as much as possible.